aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/DeleteDirStepTest.java

```diff
@@ -0,0 +1,128 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.workflow.steps;
+
+import java.io.File;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import hudson.model.queue.QueueTaskFuture;
+
+public class DeleteDirStepTest {
+
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+
+    @Test
+    public void testDeleteEmptyWorkslapce() throws Exception {
+        String workspace = runAndGetWorkspaceDir(
+                "node {\n" +
+                "  deleteDir()\n" + 
+                "}");
+        File f = new File(workspace);
+        Assert.assertFalse("Workspace driectory should no longer exist", f.exists());
+    }
+
+    @Test
+    public void testDeleteTopLevelDir() throws Exception {
+        String workspace = runAndGetWorkspaceDir(
+                "node {\n" +
+                "  writeFile file: 'f1', text: 'some text'\n" +
+                "  writeFile file: 'f2', text: 'some text'\n" +
+                "  writeFile file: '.hidden', text: 'some text'\n" +
+                "  writeFile file: 'sub1/f1', text: 'some text'\n" +
+                "  writeFile file: '.sub2/f2', text: 'some text'\n" +
+                "  writeFile file: '.sub3/.hidden', text: 'some text'\n" +
+                "  echo 'workspace is ---' + pwd() + '---'\n" + 
+                "  deleteDir()\n" + 
+                "}");
+        File f = new File(workspace);
+        Assert.assertFalse("Workspace driectory should no longer exist", f.exists());
+    }
+
+    @Test
+    public void testDeleteSubFolder() throws Exception {
+        String workspace = runAndGetWorkspaceDir(
+                "node {\n" +
+                "  writeFile file: 'f1', text: 'some text'\n" +
+                "  writeFile file: 'f2', text: 'some text'\n" +
+                "  writeFile file: '.hidden', text: 'some text'\n" +
+                "  writeFile file: 'sub1/f1', text: 'some text'\n" +
+                "  writeFile file: '.sub2/f2', text: 'some text'\n" +
+                "  writeFile file: '.sub3/.hidden', text: 'some text'\n" +
+                "  echo 'workspace is ---' + pwd() + '---'\n" + 
+                "  dir ('sub1') {\n" +
+                "    deleteDir()\n" +
```

Comment:
It's ok like it is, just a matter of usability.
I think it should use the context as it does now (`@StepContextParameter private FilePath cwd`) and accept a parameter to append to `cwd`, so the `dir ('sub1') {...}` stuff could be omitted.

It can be done in a future PR, of course.